Classes/Updater.lua

```diff
@@ -11,129 +11,65 @@ cUpdater = {}
 
 
 
-local g_CheckAttempts = 0
-local g_DownloadAttempts = 0
-local g_ReceivedPluginInfo = ""
-local g_LatestVersion = nil
-local g_WorldEditZip = nil
+cUpdater.s_CheckAttempts = 0
+cUpdater.s_DownloadAttempts = 0
 
 
 
 
-
-local VersionCheckForNewerVersionerCallbacks =
-{
-	OnConnected = function(a_TCPLink)
-		local res, msg = a_TCPLink:StartTLSClient()
-		a_TCPLink:Send("GET /cuberite/WorldEdit/master/Info.lua HTTP/1.0\r\nHost: raw.githubusercontent.com\r\n\r\n")
-	end,
-	
-	OnError = function(a_TCPLink, a_ErrorCode, a_ErrorMsg) 
-		LOGWARNING("Error while checking for newer WorldEdit version: " .. a_ErrorMsg)
-	end,
-	
-	OnReceivedData = function(a_TCPLink, a_Data)
-		g_ReceivedPluginInfo = g_ReceivedPluginInfo .. a_Data
-	end,
-	
-	OnRemoteClosed = function(a_TCPLink)
-		cUpdater:ParseResults()
-	end,
-}
-
-
-
-
-
-local NewVersionDownloaderCallbacks =
-{
-	OnConnected = function(a_TCPLink)
-		-- Start a secured connection
-		local res, msg = a_TCPLink:StartTLSClient("", "", "")
-		
-		-- Send Request Header
-		a_TCPLink:Send("GET /cuberite/WorldEdit/zip/master HTTP/1.0\r\nHost: codeload.github.com\r\nConnection: Keep-alive\r\n\r\n")
-		
-		-- Clear data we might have from previous attempts
-		g_WorldEditZip = ""
-	end,
-	
-	OnError = function(a_TCPLink, a_ErrorCode, a_ErrorMsg)
-		if (g_DownloadAttempts < g_Config.Updates.NumAttempts) then
-			g_DownloadAttempts = g_DownloadAttempts + 1
-			cUpdater:DownloadLatestVersion()
-		else
-			LOGWARNING("Error while downloading newer worldedit version: " .. a_ErrorCode)
-		end
-	end,
-	
-	OnReceivedData = function(a_TCPLink, a_Data)
-		-- Append the received Data to the saved data
-		g_WorldEditZip = g_WorldEditZip .. a_Data
-	end,
+function cUpdater:CheckForNewerVersion()
+	if (cUpdater.s_CheckAttempts > g_Config.Updates.NumAttempts) then
+		LOGWARNING("Could not connect to Github to check for a newer version for WorldEdit")
+		return;
+	end
+	cUpdater.s_CheckAttempts = cUpdater.s_CheckAttempts + 1;
 	
-	OnRemoteClosed = function(a_TCPLink)
-		-- Split the data and the headers
-		local Header, Data = g_WorldEditZip:match("^(.-)\r\n\r\n(.*)")
-		
-		-- Parse the headers so we can actually use them to validate the data
-		Header = cUpdater:ParseHeaders(Header)
-		
-		-- Check if the received data size matches that that was expected
-		if (Data:len() ~= tonumber(Header["content-length"] or -1)) then
-			if (g_DownloadAttempts < g_Config.Updates.NumAttempts) then
-				g_DownloadAttempts = g_DownloadAttempts + 1
-				cUpdater:DownloadLatestVersion()
+	cUrlClient:Get("https://raw.githubusercontent.com/cuberite/WorldEdit/master/Info.lua",
+		function(a_Body, a_Data)
+			if (a_Body) then
+				
+				cUpdater:ParsePluginInfo(a_Body)
 			else
-				LOGWARNING("Error while downloading newer worldedit version")
+				cUpdater:CheckForNewerVersion()
```

Comment:
Perhaps add a comment - `The request failed, retry`. You could also schedule a task onto the default world to retry later.